aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -297,14 +306,19 @@ impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
         resource_tag: &Self::ResourceTag,
         maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<Bytes>> {
-        // TODO: resource_group_write_set also contains a layout. What to do with it?
-        if let Some((write_op, _layout)) = self
+        if let Some((write_op, layout)) = self
             .change_set
             .resource_group_write_set()
             .get(group_key)
             .and_then(|g| g.inner_ops().get(resource_tag))
         {
-            Ok(write_op.extract_raw_bytes())
+            if layout_matches(maybe_layout, layout.as_deref()) {
+                Ok(write_op.extract_raw_bytes())
+            } else {
+                //TODO[agg_v2](fix): What should be the error here?
```

Comment:
Why not bail!("Provided layout {:?} is different from recorded {:?}"), and also log an error (here or maybe caller will)? Returning Ok(None) seems the worst we can do even in the interim for testing.